logstash-core/lib/logstash/util/cloud_setting_auth.rb

```diff
@@ -0,0 +1,29 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/util/password"
+
+module LogStash module Util class CloudSettingAuth
+  attr_reader :original, :username, :password
+
+  def initialize(value)
+    return if value.nil?
+
+    unless value.is_a?(String)
+      raise ArgumentError.new("Cloud Auth must be String. Received: #{value.class}")
+    end
+    @original = value
+    @username, sep, password = @original.partition(":")
+    if @username.empty? || sep.empty? || password.empty?
+      raise ArgumentError.new("Cloud Auth username and password format should be \"<username>:<password>\".")
+    end
+    @password = LogStash::Util::Password.new(password)
+  end
+
+  def to_s
+    "#{@original}:#{@password}"
+  end
+
+  def inspect
+    to_s
+  end
+end end end
```

Comment:
because its a utility class? While these two files are currently used for modules I felt that they are a kind of validating value object used by settings (validating) and modules (value object)